pkg/apis/federation/v1alpha1/federatedserviceoverride_types.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"log"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/endpoints/request"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/marun/federation-v2/pkg/apis/federation"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// FederatedServiceOverride
+// +k8s:openapi-gen=true
+// +resource:path=federatedserviceoverrides,strategy=FederatedServiceOverrideStrategy
+type FederatedServiceOverride struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   FederatedServiceOverrideSpec   `json:"spec,omitempty"`
+	Status FederatedServiceOverrideStatus `json:"status,omitempty"`
+}
+
+// FederatedServiceOverrideSpec defines the desired state of FederatedServiceOverride
+type FederatedServiceOverrideSpec struct {
+	Overrides []FederatedServiceClusterOverride
+}
+
+// FederatedServiceClusterOverride defines the overrides for a named cluster
+type FederatedServiceClusterOverride struct {
+	// TODO(marun) Need to ensure that a cluster name only appears
+	// once.  Why can't maps be used so this validation is automatic?
+	ClusterName string         `json:"clustername,omitempty"`
+	Override    corev1.Service `json:"template,omitempty"`
```

Comment:
As previously discussed, I don't think it's desirable to define an override as the entire type.  Are there specific fields you think a user would want to override?  If not, consider avoiding implementing an override type at this time (it is optional in the sync controller).

Reply:
oh, I was assuming it was mandatory for the sync controller. I will remove the override type for service for now. Thanks for updating me.